src/EditorFeatures/CSharpTest/PullMemberUp/CSharpPullMemberUpTests.cs

```diff
@@ -2292,6 +2291,379 @@ void C([||])
+        [WorkItem(35180, "https://github.com/dotnet/roslyn/issues/35180")]
+        public async Task TestMissingRefactoringSelectionAttributeList2()
+        {
+            var testText = @"
+using System;
+
+namespace PushUpTest
+{
+    class TestAttribute : Attribute { }
+    class Test2Attribute : Attribute { }
+    public class A
+    {
+    }
+
+    public class B : A
+    {
+        [|[Test]|]
+        [Test2]
+        void C()
+        {
+        }
+    }
+}";
+            await TestQuickActionNotProvidedAsync(testText);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsPullMemberUp)]
+        [WorkItem(35180, "https://github.com/dotnet/roslyn/issues/35180")]
+        public async Task TestMissingRefactoringSelectAttributeList()
+        {
+            var testText = @"
+using System;
+
+namespace PushUpTest
+{
+    class TestAttribute : Attribute { }
+    public class A
+    {
+    }
+
+    public class B : A
+    {
+        [|[Test]|]
+        void C()
+        {
+        }
+    }
+}";
+            await TestQuickActionNotProvidedAsync(testText);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsPullMemberUp)]
+        [WorkItem(35180, "https://github.com/dotnet/roslyn/issues/35180")]
+        public async Task TestRefactoringCaretLocAfterAttributes1()
+        {
+            var testText = @"
+using System;
+
+namespace PushUpTest
+{
+    class TestAttribute : Attribute { }
+    public class A
+    {
+    }
+
+    public class B : A
+    {
+        [Test]
+        [||]void C()
+        {
+        }
+    }
+}";
+            var expected = @"
+using System;
+
+namespace PushUpTest
+{
+    class TestAttribute : Attribute { }
+    public class A
+    {
+        [Test]
+        void C()
+        {
+        }
+    }
+
+    public class B : A
+    {
+    }
+}";
+
+            await TestInRegularAndScriptAsync(testText, expected);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsPullMemberUp)]
+        [WorkItem(35180, "https://github.com/dotnet/roslyn/issues/35180")]
+        public async Task TestRefactoringCaretLocAfterAttributes2()
+        {
+            var testText = @"
+using System;
+
+namespace PushUpTest
+{
+    class TestAttribute : Attribute { }
+    class Test2Attribute : Attribute { }
+    public class A
+    {
+    }
+
+    public class B : A
+    {
+        [Test]
+        // Comment1
+        [Test2]
+        // Comment2
+        [||]void C()
+        {
+        }
+    }
+}";
+            var expected = @"
+using System;
+
+namespace PushUpTest
+{
+    class TestAttribute : Attribute { }
+    class Test2Attribute : Attribute { }
+    public class A
+    {
+        [Test]
+        // Comment1
+        [Test2]
+        // Comment2
+        void C()
+        {
+        }
+    }
+
+    public class B : A
+    {
+    }
+}";
+
+            await TestInRegularAndScriptAsync(testText, expected);
+        }
```

Comment:
what about selecting the member *not* including the attributes?  not sure if i saw a test for that.  if we have such a test, great.

Reply:
Added.